src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -190,18 +191,84 @@ ExpressionSyntax initializer
 				immutable = false;
 			}
 
-			// TODO: narrow inspection with initializer
-			//       possibly a narrower type (using initializers type) and
-			//       possibly ImmutableTypeKind.Instances if its a new T( ... ).
+			var stuff = GetStuffToCheckForMember( type, typeSyntax, initializer );
 
-			if( !m_context.IsImmutable( type, ImmutableTypeKind.Total, typeSyntax.GetLocation(), out var diagnostic ) ) {
+			// null is a signal that there is nothing fruther that needs to be checked
+			if ( stuff == null ) {
+				return immutable;
+			}
+
+			var (typeToCheck, checkKind, diagnosticLocation) = stuff.Value;
+
+			if( !m_context.IsImmutable( typeToCheck, checkKind, diagnosticLocation, out var diagnostic ) ) {
 				diagnosticSink( diagnostic );
 				immutable = false;
 			}
 
 			return immutable;
 		}
 
+		/// <summary>

```

Comment:
This stuff is also basically copy-and-paste from the MutabilityInspector, of course.